chromadb/telemetry/opentelemetry/__init__.py

```diff
@@ -129,4 +139,4 @@ def add_attributes_to_current_span(
     if not tracer:
         return
     span = trace.get_current_span()
-    span.set_attributes(_transform_attributes(attributes))  # type: ignore
+    span.set_attributes(_transform_attributes(attributes))
```

Comment:
Ah, this was sloppy on my part. The correct fix here is to simply remove the call to `_transform_attributes`.

The call to `_transform_attributes` is a holdover from when `add_attributes_to_current_span` accepted a `Dict[str, Any]`, since OTel wants attribute values to be one of `[str, bool, float, int]`. @HammadB and I talked about this in code review and decided we should strongly type `add_attributes_to_current_span` so transformation is no longer necessary.

All that said: let's delete `_transform_attributes` and the call to it.

Reply:
mypy is still complaining:

![image](https://github.com/chroma-core/chroma/assets/1157440/0ccc3d2b-ad51-4604-b892-339d5b4566d2)


Should I do this instead?:

```py
span.set_attributes({k: cast(types.AttributeValue, v)
                        for k, v in attributes.items()})
```

It seems that Otel expects `AttributeValue` so we might have to do the conversion.

Note: We can even widen the accepted values to include sequences of the ones you mentioned above:

```py
AttributeValue = Union[
    str,
    bool,
    int,
    float,
    Sequence[str],
    Sequence[bool],
    Sequence[int],
    Sequence[float],
]
```

Reply:
Ah, this is why I shouldn't review code on Sunday night ;)

I like your original method of solving this, or I'd be happy to cast to `AttributeValue` inline in `add_attributes_to_current_span` -- up to you. And sorry to have asked you to make an unnecessary change.

Reply:
Take a look at the latest change, it works and feels more minimalistic. (we live and die with mypy)